### ISSUE TITLE
Create unified state for each Suggest method

### ISSUE THREAD

**Issue by drittich** [TricksterCards/TricksterBots#24]
As I think about implementing enhancements to the EuchreBot class, I noticed there is some state information available to some `Suggest*` methods that is not available to others, as well as some state information possibly missing overall. While I could work around some of this, it seems like something that you might want to address first in the backend that provides the data. 

Are there any plans underway for this already? I can provide some of the data I am looking for, if needed. (It's possible it's there and I'm just not finding it.) I'm thinking it would make sense to unify the state into one `SuggestState` class, rather than having a separate state class for Bid, Pass, Discard, etc.

**Comment by johnson520** [comment]
The four `Suggest*State` classes have the information that the original bots used. We consciously keep the information to a minimum so as to avoid creating bots that can cheat.

If there's specific information needed for a _suggest_ method enhancement, we're open to adding it provided that it would be available to a human player playing the same seat.

**Comment by drittich** [comment]
Got it, I'll provide some examples of what I'm looking for shortly.

**Comment by drittich** [comment]
Scenario 1: In euchre with call for best enabled, when your partner orders you up and goes alone, `SuggestPass` currently passes the highest trump card. This rule should stay in place, except for the case where the card ordered up is now your highest trump, and you also hold the trump card one rank below that. In that case, it's fairly common practice to pass the card ranked one below the ordered-up trump, so your partner has more knowledge of who holds what trump that they don't.

To enable this, `SuggestPass` needs knowledge of the card that was ordered up, which I don't currently see available. It may be that the Card class already contains this info. I do not know, for example, what `IsHonorRank` means.

**Comment by johnson520** [comment]
When the partner of the dealer goes alone, the upcard has already been added to their hand when SuggestPass is invoked. This allows issue #30 to be implemented without a change to the SuggestPassState.

**Comment by drittich** [comment]
How do we determine which card in the hand was the up card?

**Comment by antross** [comment]
I agree, we'd need the up card added to the suggest pass state to implement this enhancement (either that or some generic knowledge of which cards in our hand are already "known" to other players).
